client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -117,9 +119,11 @@ const Header: FunctionComponent< Props > = ( {
 				</div>
 			</div>
 			<div className="gutenboarding__header-section">
-				<Link to={ next } className="gutenboarding__header-next-button" isPrimary isLarge>
-					{ NO__( 'Next' ) }
-				</Link>
+				{ ! isNextHidden && (
```

Comment:
How about we change this to

```suggestion
				{ ! next && (
```

?

Falsey `next` can indicate that there's no next step to go to, and it'll allow us to drop the extra bool prop. 

(I'm generally a bit wary of bool props controlling visibility of elements, since we might easily end up with an unwieldy number of them, especially in a context like Gutenboarding, which is all about a 'progressive chrome' that adds more and more UI elements as the user progresses through the steps of the onboarding block.)

Reply:
@ockham thanks I wasn't too sure about adding the extra prop. In your suggestion, I think you mean we'd check that `next` is truthy to render the button?

The only problem is that checking if `next` is truthy will hide the disabled next button on the design step, too. Another way we could do it would be to set the value of `next` to a constant like `'HIDE_BUTTON'` and if that's the value, then hide the button. What do you think?

Reply:
I implemented it that way at first in https://github.com/Automattic/wp-calypso/pull/38705/commits/e548cae1f7931d746795f7738fbe6458926c99a0 but then second guessed myself :)

Reply:
> The only problem is that checking if next is truthy will hide the disabled next button on the design step, too.

I'm a bit unsure of myself, but I'll ask anyhow. Under the current architecture, wouldn't the design step define it's own `next` in the switch block (thereby showing the next button)? 

```
		case Step.IntentGathering:
				next = null;
			break;
        case Step. DesignSelection:
				next = Step. SomeFabulousStep;
			break;
```

Or is it more to do with the disabled state?

Reply:
Good point, I think it would. I'm possibly over thinking it, but I was mostly thinking about how do we want to handle the disabled state. So we've got 3 potential states for the button: hidden, disabled, and active link.

Reply:
> So we've got 3 potential states for the button: hidden, disabled, and active link.

Got it. I suppose we'll want to update the button based on validating input values in the main section or some such thing in the future.

Reply:
> @ockham thanks I wasn't too sure about adding the extra prop. In your suggestion, I think you mean we'd check that `next` is truthy to render the button?

Yeah, you're right :sweat_smile: 

> The only problem is that checking if `next` is truthy will hide the disabled next button on the design step, too.

That's a good point.

> Another way we could do it would be to set the value of `next` to a constant like `'HIDE_BUTTON'` and if that's the value, then hide the button. What do you think?

Let's avoid magic constants :grimacing: They're a bit too obscure, and go a bit against the whole types paradigm.

---

It's fine to go with either the current state of the PR (that checks `next`), or to revert to your original `isNextHidden` approach. My apologies for delaying this PR, we probably shouldn't overthink it at this stage.

Just to explain why I was skeptical about `isNextHidden` -- it's mostly about how this pattern will scale:

One of Gutenboarding's paradigms is a 'progressive chrome' -- the editor (with its header and possibly sidebar) adds more and more UI controls, as the user progresses through the onboarding steps in the onboarding block.

With the `isSomethingHidden` bool pattern, we'd end up with a number of bool props that control which of those UI elements to conditionally hide, depending on which step we're currently at. This is exacerbated by the fact that those UI controls might dynamically change (e.g. the 'Next' button label might be different at different stages).

I don't have a clear answer how to best solve this right now, but I'm leaning towards 'regions' components that accept a number of component props to be placed in different parts of the UI. See e.g. https://github.com/WordPress/gutenberg/blob/a8f43fa7e21a028b43d0cb023ac5c872a3fbd4d8/packages/edit-post/src/components/editor-regions/index.js (introduced by https://github.com/WordPress/gutenberg/pull/18044, which has some explanation in its PR desc).

One problem that's a bit different for our use case is encapsulation -- I don't think we want `gutenboarding.tsx` to pass all those granular components down to e.g. the header or sidebar; instead, those 'container'/'region' components should infer themselves which components to display at what stage. Some of that information can be inferred from the onboarding data store; we'll still have to consider how to make those components aware of routing, tho (maybe we'll end up moving our main routing logic out of the onboarding block -- I don't know at this point). Note that this is basically a question about which pattern we'll use to build the 'custom editor'.

Anyway, I won't block this PR any longer :sweat_smile: Let's just keep this in mind as we add more 'conditional' UI elements.